mods/fire/init.lua

```diff
@@ -95,38 +95,18 @@ minetest.register_tool("fire:flint_and_steel", {
 		itemstack:add_wear(1000)
 		if pt.type == "node" then
 			local node_under = minetest.get_node(pt.under).name
-			local is_coalblock = node_under == "default:coalblock"
-			local is_tnt = node_under == "tnt:tnt"
-			local is_gunpowder = node_under == "tnt:gunpowder"
-			if minetest.get_item_group(node_under, "flammable") >= 1 or
-					is_coalblock or is_tnt or is_gunpowder then
-				local flame_pos = pt.above
-				if is_coalblock then
-					flame_pos = {x = pt.under.x, y = pt.under.y + 1, z = pt.under.z}
-				elseif is_tnt or is_gunpowder then
-					flame_pos = pt.under
-				end
-				if minetest.get_node(flame_pos).name == "air" or
-						is_tnt or is_gunpowder then
-					local player_name = user:get_player_name()
-					if not minetest.is_protected(flame_pos, player_name) then
-						if is_coalblock then
-							minetest.set_node(flame_pos,
-								{name = "fire:permanent_flame"})
-						elseif is_tnt then
-							minetest.set_node(flame_pos,
-								{name = "tnt:tnt_burning"})
-						elseif is_gunpowder then
-							minetest.set_node(flame_pos,
-								{name = "tnt:gunpowder_burning"})
-						else
-							minetest.set_node(flame_pos,
-								{name = "fire:basic_flame"})
-						end
-					else
-						minetest.chat_send_player(player_name, "This area is protected")
-					end
-				end
+			local nodedef = minetest.registered_nodes[node_under]
+			if not nodedef then
+				return
+			end
+
+			local player_name = user:get_player_name()
+			if nodedef.on_ignite
+					and not minetest.is_protected(pt.under, player_name) then
+				nodedef.on_ignite(pt.under, user)
+			elseif minetest.get_item_group(node_under, "flammable") >= 1
+					and not minetest.is_protected(pt.above, player_name) then
```

Comment:
Ah, I think you meant that in the case a flammable node has an on_ignite, if both positions are protectewd both will be checked.

Yes, this was like needed because they were different positions. If they are the same position then it should be factored out, of course.